Tests/StressTests/Support/CollectionViewUpdater.swift

```diff
@@ -0,0 +1,169 @@
+    // MARK: - State
+    private var collection = Collection() // for UICollectionViewDataSource
+    private var collectionView: UICollectionView?
+    private var window: UIWindow?
+    
+    // MARK: - Internal
+    func updateCollectionView(
+        from: Collection,
+        to: Collection,
+        completion: @escaping (CollectionViewUpdaterResult) -> ())
+    {
+        let collectionView = makeAndReloadCollectionViewWith(collection: from)
+        
+        let diff = StagedChangeset(
+            source: from,
+            target: to
+        )
+        
+        performBatchUpdates(
+            of: collectionView,
+            collection: to,
+            diff: diff,
+            completion: completion
+        )
+    }
+    
+    func cleanUp() {
+        collection = []
+        collectionView?.dataSource = nil
+        collectionView?.removeFromSuperview()
+        collectionView = nil
+        window = nil
+    }
+
+    // MARK: - Private
+    private func makeAndReloadCollectionViewWith(
+        collection: Collection)
+        -> UICollectionView
+    {
+        let window = UIWindow(frame: UIScreen.main.bounds)
+        window.rootViewController = UIViewController()
+        window.layer.speed = 100 // Speed up the animations
+        self.window = window
+        
+        let layout = UICollectionViewFlowLayout()
+        layout.itemSize = CGSize(width: 1, height: 1) // To fit all cells in the screen 
+        layout.minimumInteritemSpacing = 0.001
+        layout.minimumLineSpacing = 0.001
+        layout.headerReferenceSize = CGSize(width: 1, height: 1) // To fit all headers in the screen 
+        
+        let collectionView = UICollectionView(frame: window.frame, collectionViewLayout: layout)
+        self.collectionView = collectionView
+        window.addSubview(collectionView)
+        
+        self.collection = collection
+        collectionView.dataSource = self
+        collectionView.register(Cell.self, forCellWithReuseIdentifier: Cell.reuseIdentifier)
+        collectionView.register(Section.self, forSupplementaryViewOfKind: Section.kind, withReuseIdentifier: Section.reuseIdentifier)
+        collectionView.reloadData()
+        
+        return collectionView
+    }
+    
+    private func performBatchUpdates(
+        of collectionView: UICollectionView,
+        collection: Collection,
+        diff: StagedChangeset<Collection>,
+        completion: @escaping (CollectionViewUpdaterResult) -> ())
+    {
+        guard !diff.isEmpty else {
+            // If we apply an empty diff to a collection view, then for some reason
+            // `collectionView.indexPathsForVisibleItems` and `collectionView.visibleCells` return empty arrays,
+            // even if `collectionView.numberOfItems(inSection:)` returns a non empty number.
+            // So we cannot perform normal assertions as in `CollectionViewUpdaterResult.success`
+            return completion(.noUpdateRequired)
```

Comment:
This is a strange case. I have no idea, why this happens.
Same situation on `ListDiff` stress tests did not take place.
@ra1028, Do you have any thoughts about it?

Reply:
Hmm, I'm not sure.
Nothing should happen in the empty StagedChangeset.